File: class 2/10866.py

class deque() :
    def __init__(self) :
        self.deque= []
    
    def push_front(self, data) :
        self.deque.insert(0, data)

    def push_back(self, data) :
        self.deque.append(data)

    def pop_front(self) :
        pop_object = None

        if self.isEmpty() :
            pop_object = -1     
        else :
            pop_object = self.deque[0]
            self.deque = self.deque[1:]

        return pop_object

    def pop_back(self) :
        pop_object = None

        if self.isEmpty() :
            pop_object = -1     
        else :
            pop_object = self.deque[-1]
            self.deque.pop()

        return pop_object



    def size(self) :
        return len(self.deque)

    def isEmpty(self) :
        is_empty = False 
        if len(self.deque) == 0 :
            is_empty = True
        
        return is_empty


    def empty(self) :
        is_empty = None
        if len(self.deque) == 0 :
            is_empty = 1
        else :
            is_empty = 0
        
        return is_empty

    def front(self) :
        front_object = None

        if self.isEmpty() :
            front_object = -1
        else :
            front_object = self.deque[0]
        
        return front_object

    def back(self) :
        back_object = None

        if self.isEmpty() :
            back_object = -1
        else :
            back_object = self.deque[-1]
        
        return back_object
    
import sys    

num = int(sys.stdin.readline())
q = deque()

for i in range(num) :
    x = sys.stdin.readline().split()

    if x[0] == 'push_back' :
        q.push_back(int(x[1]))
    
    elif x[0] == 'push_front' :
        q.push_front(int(x[1]))

    elif x[0] == "pop_front" :
        print(q.pop_front())

    elif x[0] == "pop_back" :
        print(q.pop_back())

    elif x[0] == "size" :
        print(q.size())
    
    elif x[0] == "empty" :
        print(q.empty())
    
    elif x[0] == "front" :
        print(q.front())
    
    elif x[0] == "back" :
        print(q.back())
        
    

        

File: 백준 문제 풀이 모음/kotlin/class 2/10866.kt
fun main() = with(System.`in`.bufferedReader()) {
    val n = readLine().toInt()

    val deque = ArrayDeque<Int>()
    val sb = StringBuilder()
    repeat(n) {
        val input = readLine().split(' ').toList()
        if (input.size == 2) {
            if (input[0].equals("push_back")) {
                deque.addLast(input[1].toInt())
            } else if (input[0].equals("push_front")) {
                deque.addFirst(input[1].toInt())
            }
        } else {
            if (input[0].equals("size")) {
                sb.append(deque.size)
                sb.append('\n')
            } else if(input[0].equals("empty")) {
                if (deque.isEmpty()) {
                    sb.append(1)
                } else {
                    sb.append(0)
                }
                sb.append('\n')
            } else if(input[0].equals("front")) {
                if (deque.isEmpty()) {
                    sb.append(-1)
                } else {
                    sb.append(deque.first())
                }
                sb.append('\n')
            } else if (input[0].equals("back")) {
                if (deque.isEmpty()) {
                    sb.append(-1)
                } else {
                    sb.append(deque.last())
                }
                sb.append('\n')
            } else if (input[0].equals("pop_front")) {
                if (deque.isEmpty()) {
                    sb.append(-1)
                } else {
                    sb.append(deque.removeFirst())
                }
                sb.append('\n')
            } else if (input[0].equals("pop_back")) {
                if (deque.isEmpty()) {
                    sb.append(-1)
                } else {
                    sb.append(deque.removeLast())
                }
                sb.append('\n')
            }
        }
    }
    println(sb.toString())
}